### ISSUE TITLE
Fine grain classifier specification on cli

### ISSUE THREAD

**Issue by wisechengyi** [coursier/coursier#717]
Currently `--classifier` is applied to all modules requested on the cli. Ideally we should be able specify below in the same run.
```
fetch org:name:version \ # default classfieir
org2:name2:version2:classifierX
```
The same may also apply to packaging.

Quote from @baroquebobcat:

It looks like you can provide coursier with classifiers via the extra attribute syntax. I tried it locally with org.apache.kafka:kafka-streams:1.0.0;classifier=test, and it was accepted, but ignored. Maybe that's something we could file an issue about. I think it would clean this up a lot, and also allow projects with classified dependencies to function.

https://github.com/coursier/coursier/blob/8f0fef6039f1eb6321997446ab565020f9b8af97/core/shared/src/main/scala/coursier/util/Parse.scala#L24

**Comment by wisechengyi** [comment]
@alexarchambault I plan to work on this soon. Let me know if you have any preference in term of 
`org2:name2:version2:<classifier>` vs `org2:name2:version2;classifier=<classifier>`, or some other format.

I think this should be the last blocking issue for us to fully adopt coursier internally.

**Comment by alexarchambault** [comment]
Let's go for the latter, the first one conflicts with [the one for configurations](https://github.com/coursier/coursier/blob/70fa5f13cb36b2e30bfdddbc76566cbf878bbd79/core/shared/src/main/scala/coursier/util/Parse.scala#L130).

**Comment by wisechengyi** [comment]
Sounds good.

**Comment by wisechengyi** [comment]
@axarchambault how do feel about using something other than `;` because it is the separator for bash, so the behavior may be confusing for user who misses quoting it.

How about `#` or `$`? E.g. `org2:name2:version2#classifier=<classifier>`, `org2:name2:version2$classifier=<classifier>`

**Comment by alexarchambault** [comment]
Maybe we could use `:` again… Like `org:name:version:a=b:c=d` To avoid conflicts with `org:name:version:<classifier>`, we can check for `=` in the forth part.

**Comment by wisechengyi** [comment]
Using `:` is fine. User should be still able to specify config if they want along with classifier, and I do not intend to change the match cases here:
https://github.com/coursier/coursier/blob/74b2f33dd67d0a0ef8e40236fc9ebedeb09113ef/core/shared/src/main/scala/coursier/util/Parse.scala#L164-L187

so both should be valid:
org:name:version:a=b:c=d
org:name:version:config:a=b:c=d

What we can do is to enforce the order, say attributes (a.k.a keyword args) have to come after regular args. Do you think that could work? @alexarchambault

**Comment by wisechengyi** [comment]
Ended up using `::` to separate args from kwargs in #735 because it's relatively easy to implement. E.g.
`org:name:version::a=b::c=d`,
`org:name:version:config::a=b::c=d`

Also order is enforced, so `org:name:version::a=b::config::c=d` is illegal. Please let me know if you have a strong opinion against it.

**Comment by wisechengyi** [comment]
spoke too soon... `::` has already some meanings https://github.com/coursier/coursier/blob/master/core/shared/src/main/scala/coursier/util/Parse.scala#L141

naming is hard...

**Comment by wisechengyi** [comment]
closed with #735
